Python/specialize.c

```diff
@@ -2353,6 +2353,7 @@ int
 void
 _Py_Specialize_ForIter(PyObject *iter, _Py_CODEUNIT *instr, int oparg)
 {
+    assert(_PyOpcode_Deopt[instr->op.code] == FOR_ITER);
```

Comment:
We should really add such asserts to many specialization functions; I ran into this one during an intense debugging session.